### PR TITLE
Fixes #17867 - Show existing products on discovery create

### PR DESCRIPTION
**Pull request by dLobatog**
The create products after discovery page does not use the styles
described on
http://www.patternfly.org/pattern-library/forms-and-controls/forms/#_

Another problem it has is that even when there are no products
available, it shows you an option 'existing product' but clicking on it
does nothing at all

screenshot from 2016-12-28 11-46-53

### CODE REVIEW

**Comment by ehelms** [comment]
Can oyu fix the HTML indentation to two spaces?

**Comment by ehelms** [comment]
I'd also consult with @waldenraines whether this should adopt to his changes around using horizontal forms.

I tend to like having options around even if they are empty, otherwise I think users tend not to know they even exist. So if we hide the ability to use an existing product just because there aren't currently, users may or may not learn they could use an existing product (or might ask themselves, why isn't my product showing up -- do I not have permissions? Is there an error ?)

**Comment by dLobatog** [comment]
Fixed the indentation and now the form uses bst-form-group from Bastion (like most other Katello forms). I removed the `if condition` about "exisiting products" as that seems broken to me, no matter if I have custom content or not, it never shows up.

Errors didn't display properly prior to this change, I modified the controller so they do now.

**Comment by dLobatog** [comment]
screenshot from 2016-12-28 18-08-38

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html [hunk 1]
```diff
@@ -1,91 +1,89 @@
 <span page-title>{{ 'Create Repositories' | translate }}</span>
 
-<form name="productForm" class="col-sm-5" ng-disabled="creating()" novalidate role="form">
+<form name="productForm" class="col-sm-5 form-horizontal" ng-disabled="creating()" novalidate role="form">
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: We are moving away from using `.form-horizontal`.  Do you mind changing this to use the normal bs3 form?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html [hunk 2]
```diff
@@ -95,48 +93,39 @@ <h4 ng-show="discovery.selected.length > 0" translate>Selected URLs:</h4>
...
+          class="form-control"
+          required/>
         </div>
       </div>
     </div>
+    <button ng-click="createRepos()"
+      ng-disabled="creating() || !requiredFieldsEnabled()"
+      translate
+      class="btn btn-primary">
+      <i class="pficon-add-circle-o inline-icon"></i>	Create
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: Can you remove this icon?  We are moving towards not having icons in the buttons.

Comment (dLobatog) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: No problem - I added the icon because of https://groups.google.com/forum/?fromgroups#!topic/foreman-dev/jy2Ytas4SRw , @rohoover any idea what to do?

Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: Ah okay, I just felt it was superfluous but I don't have too strong of an opinion about it.

**Comment by dLobatog** [comment]
Repushed:

* I removed form-horizontal
* The controller now uses isUndefined to comply with eslint rules

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html (hunk 2 above)
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: I don't want to hold this PR up on one icon.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html [hunk 3]
```diff
@@ -3,89 +3,87 @@
...
     </div>
 
-    <div class="form-group">
-      <div class="col-md-5" ng-hide="createRepoChoices.newProduct">
-        <select class="form-control" ng-disabled="creating()"
-                ng-model="createRepoChoices.existingProductId"
-                ng-options="product.id as product.name for product in products">
-        </select>
+
+    <div class="clearfix">
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: Why is this needed?

**Comment by dLobatog** [comment]
@waldenraines No reason at all for the various 'clearfixes' , thanks for catching that. Notice the list of existing products does *not* show up either with or without this change, I'm trying to debug the form controller but still I'm not clear what's going on (Angular newbie here)

**Comment by dLobatog** [comment]
@waldenraines Fixed the issue with showing existing products: ng-hide couldn't work because it was checking on newProduct === `true` (boolean) but `"true"` is a string in the discovery-form controller. Those were all the changes I wanted to make. Thanks for reviewing

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html [hunk 4]
```diff
@@ -3,89 +3,76 @@
...
+        ng-model="createRepoChoices.newProduct"
+        ng-disabled="creating()"/>
         <span translate>New Product</span>
       </label>
     </div>
 
-    <div class="form-group">
-      <div class="col-md-5" ng-hide="createRepoChoices.newProduct">
+
+    <div ng-hide="createRepoChoices.newProduct === 'true'">
```
Comment (dLobatog) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: This didn't work before because newProduct is a string, not a boolean

Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: Is there a way to cast it to a boolean?  This seems rather fragile to me.

Comment (dLobatog) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html]: @waldenraines The problem is that `newProduct` comes from the HTML tag value at lines 14 and 23. Those are sent as strings to angular and I don't think they can be sent as booleans. 

To auto parse them as booleans I could write a directive like http://jsfiddle.net/zk99k4fg/1/ but I think it's overkill in this case (what do we gain from checking the string vs a boolean in this case?)

**Comment by dLobatog** [comment]
@waldenraines Replied inline to the issue of `newProduct` being a 'true/false' string vs boolean.

**Comment by dLobatog** [comment]
@waldenraines Anything I'm missing here?

**Comment by waldenraines** [comment]
> @waldenraines Anything I'm missing here?

No, sorry I missed this, approved now.

**Comment by jlsherrill** [comment]
Thanks @dLobatog !
